tests/applications/test_junebug_message_api.py

```diff
@@ -0,0 +1,667 @@
+            await http_server.send_rsp(RspInfo())
+
+    assert req.path == "event"
+    assert req.headers["Content-Type"] == "application/json"
+    assert req.headers["Authorization"] == "Token my-event-token"
+    assert req.body_json["event_type"] == "submitted"
+    assert req.body_json["event_details"] == {}
+    assert req.body_json["channel_id"] == "jma-test"
+
+
+async def test_forward_ack_bad_response(jma_worker, http_server, caplog):
+    """
+    If forwarding an ack results in an HTTP error, the error and event
+    are logged.
+    """
+    await store_ehi(jma_worker, "msg-21", f"{http_server.bind}/event", None)
+    ev = mkev("msg-21", EventType.ACK)
+
+    with fail_after(2):
+        async with handle_event(jma_worker, ev):
+            await http_server.receive_req()
+            await http_server.send_rsp(RspInfo(code=500))
+
+    [err] = [log for log in caplog.records if log.levelno >= logging.ERROR]
+    assert "Error sending event, received HTTP code 500" in err.getMessage()
+
+
+async def test_forward_nack(jma_worker, http_server):
+    """
+    A nack event referencing an outbound message we know about is
+    forwarded over HTTP.
+    """
+    await store_ehi(jma_worker, "msg-21", f"{http_server.bind}/event", None)
+    ev = mkev("msg-21", EventType.NACK, nack_reason="KaBooM!")
+
+    with fail_after(2):
+        async with handle_event(jma_worker, ev):
+            req = await http_server.receive_req()
+            await http_server.send_rsp(RspInfo())
+
+    assert req.path == "event"
+    assert req.headers["Content-Type"] == "application/json"
+    assert req.body_json["event_type"] == "rejected"
+    assert req.body_json["event_details"] == {"reason": "KaBooM!"}
+    assert req.body_json["channel_id"] == "jma-test"
+
+
+async def test_forward_dr(jma_worker, http_server):
+    """
+    A delivery report event referencing an outbound message we know
+    about is forwarded over HTTP.
+    """
+    await store_ehi(jma_worker, "m-21", f"{http_server.bind}/event", None)
+    ev = mkev("m-21", EventType.DELIVERY_REPORT, delivery_status=DeliveryStatus.PENDING)
+
+    with fail_after(2):
+        async with handle_event(jma_worker, ev):
+            req = await http_server.receive_req()
+            await http_server.send_rsp(RspInfo())
+
+    assert req.path == "event"
+    assert req.headers["Content-Type"] == "application/json"
+    assert req.body_json["event_type"] == "delivery_pending"
+    assert req.body_json["event_details"] == {}
+    assert req.body_json["channel_id"] == "jma-test"
+
+
+# TODO: Tests for outbound messages. In the original junebug codebase, those
+#       are part of the API rather than the worker.
+
+## Test cases from the original junebug worker codebase. The ones that are left
+## are for things we may not want/need anymore.
+
+## Various message statistics. We don't seem to use these for rate limiting or
+## anything, so unless we're querying them somewhere we probably don't need
+## them.
```

Comment:
They were part of the original design, to be able to provide things like a dashboard with channel health, but never used, so they can be excluded from this. That sort of thing is probably better as prometheus metrics nowadays.